presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -617,4 +683,34 @@ private boolean isSchemaOwner(SystemSecurityContext context, String schemaName)
         }
         return false;
     }
+
+    private boolean checkAnyTablePermission(SystemSecurityContext context, SchemaTableName tableName)
+    {
+        return checkTablePermission(context, tableName, privileges -> !privileges.isEmpty());
+    }
+
+    private boolean checkTablePermission(SystemSecurityContext context, SchemaTableName tableName, TableAccessControlRule.TablePrivilege... requiredPrivileges)
+    {
+        return checkTablePermission(context, tableName, privileges -> privileges.containsAll(ImmutableSet.copyOf(requiredPrivileges)));
+    }
+
+    private boolean checkTablePermission(SystemSecurityContext context, SchemaTableName tableName, Predicate<Set<TableAccessControlRule.TablePrivilege>> checkPrivileges)
+    {
+        if (tableRules.isEmpty()) {
+            return true;
+        }
+
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
```

Comment:
I don't think anyone should be allowed to modify `INFORMATION_SCHEMA_NAME` tables. But I see it is how it is implemented in `FileBasedAccessControl`.

Reply:
@findepi added to #1559